recipes/dcmtk/all/conanfile.py

```diff
@@ -51,149 +52,190 @@ class DCMTKConan(ConanFile):
-
+            tc.variables["DCMTK_WITH_WRAP"] = self.options.with_tcpwrappers
+        tc.variables["DCMTK_WITH_OPENJPEG"] = False # not used at all, do not try to add an option for this one
+        tc.variables["DCMTK_ENABLE_PRIVATE_TAGS"] = self.options.builtin_private_tags
+        tc.variables["DCMTK_WITH_THREADS"] = self.options.with_multithreading
+        tc.variables["DCMTK_WITH_DOXYGEN"] = False
+        tc.variables["DCMTK_WIDE_CHAR_FILE_IO_FUNCTIONS"] = self.options.wide_io
+        tc.variables["DCMTK_WIDE_CHAR_MAIN_FUNCTION"] = self.options.wide_io
+        tc.variables["DCMTK_ENABLE_STL"] = self.options.enable_stl
+        tc.variables["DCMTK_ENABLE_CXX11"] = True
+        tc.variables["DCMTK_ENABLE_MANPAGE"] = False
+        tc.cache_variables["DCMTK_DEFAULT_DICT"] = self.options.default_dict
+        tc.variables["DCMTK_USE_DCMDICTPATH"] = self.options.use_dcmdictpath
         if self.settings.os == "Windows":
-            cmake.definitions["DCMTK_OVERWRITE_WIN32_COMPILER_FLAGS"] = False
-
+            tc.variables["DCMTK_OVERWRITE_WIN32_COMPILER_FLAGS"] = False
         if is_msvc(self):
-            cmake.definitions["DCMTK_ICONV_FLAGS_ANALYZED"] = True
-            cmake.definitions["DCMTK_COMPILE_WIN32_MULTITHREADED_DLL"] = "MD" in msvc_runtime_flag(self)
+            tc.variables["DCMTK_ICONV_FLAGS_ANALYZED"] = True
+            tc.variables["DCMTK_COMPILE_WIN32_MULTITHREADED_DLL"] = not is_msvc_static_runtime(self)
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        deps.generate()
+
+    def _collect_cmake_required(self, dependency):
+        dep = self.dependencies.host[dependency]
+        dep_cpp_info = dep.cpp_info.aggregated_components()
+        dep_includes = [p.replace("\\", "/") for p in dep_cpp_info.includedirs]
+        dep_defs = [d for d in dep_cpp_info.defines]
+        libdirs_flag = "/LIBPATH:" if is_msvc(self) else "-L"
+        dep_libdirs = ["{}{}".format(libdirs_flag, p.replace("\\", "/")) for p in dep_cpp_info.libdirs]
+        dep_libs = [l for l in dep_cpp_info.libs]
+        dep_system_libs = [s for s in dep_cpp_info.system_libs]
+        dep_frameworks = [f"-framework {f}" for f in dep_cpp_info.frameworks]
+        for _, trans_dependency in dep.dependencies.items():
+            if trans_dependency.context != "host":
+                continue
+            trans_dep_cpp_info = trans_dependency.cpp_info.aggregated_components()
+            dep_includes.extend([p.replace("\\", "/") for p in trans_dep_cpp_info.includedirs])
+            dep_defs.extend([d for d in trans_dep_cpp_info.defines])
+            dep_libdirs.extend(["{}{}".format(libdirs_flag, p.replace("\\", "/")) for p in trans_dep_cpp_info.libdirs])
+            dep_libs.extend([l for l in trans_dep_cpp_info.libs])
+            dep_system_libs.extend([s for s in trans_dep_cpp_info.system_libs])
+            dep_frameworks.extend([f"-framework {f}" for f in trans_dep_cpp_info.frameworks])
```

Comment:
It works fine and I don't want to invalidate reviews and trigger a new build, but next time I'll try something more elegant:

```python
        # Aggregate cpp_info of dependency and all its depencencies (direct and transitive)
        dep = self.dependencies.host[dependency]
        dep_cpp_info = dep.cpp_info.aggregated_components()
        for _, trans_dep in dep.dependencies.items():
            if trans_dep.context != "host":
                continue
            dep_cpp_info.merge(trans_dep.cpp_info.aggregated_components())

        # Get compile & link flags of dependency and its dependencies
        dep_includes = [p.replace("\\", "/") for p in dep_cpp_info.includedirs]
        dep_defs = [d for d in dep_cpp_info.defines]
        libdirs_flag = "/LIBPATH:" if is_msvc(self) else "-L"
        dep_libdirs = ["{}{}".format(libdirs_flag, p.replace("\\", "/")) for p in dep_cpp_info.libdirs]
        dep_libs = [l for l in dep_cpp_info.libs]
        dep_system_libs = [s for s in dep_cpp_info.system_libs]
        dep_frameworks = [f"-framework {f}" for f in dep_cpp_info.frameworks]
```